pkg/kubelet/kubelet_pods.go

```diff
@@ -1624,6 +1624,26 @@ func (kl *Kubelet) generateAPIPodStatus(pod *v1.Pod, podStatus *kubecontainer.Po
 		if err != nil {
 			klog.V(4).InfoS("Cannot get host IPs", "err", err)
 		} else {
+			if s.HostIP != "" {
+				ok, err := utilnet.IsDualStackIPStrings([]string{s.HostIP, hostIPs[0].String()})
+				if err != nil {
+					kl.recorder.Eventf(pod, v1.EventTypeWarning, "InvalidHostIPs", "host IPs are invalid %q or %q", s.HostIP, hostIPs[0].String())
+				}
+				if ok {
+					// Once CloudDualStackNodeIPs is beta, <EXPLICIT-IPV4-ADDRESS>,<EXPLICIT-IPV6-ADDRESS> can be set to --node-ip.
+					if utilnet.IPFamilyOf(hostIPs[0]) == utilnet.IPv4 {
+						// kubelet auto-detected an IPv4 node IP (hostIPs[0]), but the cloud provider wants this to be
+						// a single-stack IPv6 or dual-stack IPv6-primary node
+						kl.recorder.Eventf(pod, v1.EventTypeWarning, "HostIPsIPFamilyMismatch",
+							"set [::] or an explicit IPv4 address to --node-ip")
```

Comment:
The information that is in the comment needs to be in the Event text instead to make it understandable (after which you no longer need the comment).

``"Kubelet auto-detected an IPv4 node IP, but the cloud provider selected an IPv6 node IP; pass an explicit `--node-ip` to kubelet to fix this."``